arangod/V8Server/V8DealerFeature.cpp

```diff
@@ -1437,3 +1437,27 @@ void V8DealerFeature::shutdownContext(V8Context* context) {
 
   delete context;
 }
+
+V8ContextDealerGuard::V8ContextDealerGuard(v8::Isolate* isolate, TRI_vocbase_t* vocbase, bool allowModification)
+  : _context(nullptr)
+  , _isolate(isolate)
+  , _active(isolate ? false : true)
+{
+  if (_active) {
+    _context = V8DealerFeature::DEALER->enterContext(vocbase, allowModification);
+    if (!_context) {
+      THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "could not acquire v8 context");
```

Comment:
didn't we say "no exceptions in new code"?

Reply:
I think exceptions should still be used when something unexpected / exceptional happens. I think the guidelines try to avoid exceptions for control-flow: e.g. throwing 5000  cluster-comm exceptions, when a server is not yet reachable, which will make ´gdb´ with ´catch throw´ useless. If we do the checking everywhere, even in places that should usually work fine, we just gain a decrease in speed. There is no silver bullet :(